### ISSUE TITLE
Install rust in docker images for building wheels

### ISSUE THREAD

**Issue by jesper-friis** [SINTEF/dlite#601]
Unfortunately pydantic v2 requires rust, cargo and poetry to be available when installing it on manylinux2010 and musllinux_1_1. This means that although DLite now has support for pydantic v2, we cannot test it when building wheels for these platforms. This also means that `extra_requirements` in python/setup.py cannot be read from `requirements_full.txt` but has to be specified separately.

To address these issues, we should install rust, cargo and poetry in the docker images we use to build the DLite wheels for manylinux2010 and musllinux_1_1. 

Basically we just need to update

    .github/docker/Dockerfile-manylinux.template
    .github/docker/Dockerfile-musllinux.template

to ensure that rust, cargo and poetry are installed. Unfortunately, it doesn't seem to be that easy... I tested for manylinux2010, but rust and cargo aren't available as yum packages on this platform and the official `curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh`  to work either.

When we somehow have figured out how to install rust and cargo, poetry should be pip installable.

**Comment by jesper-friis** [comment]
**Update**: if it is difficult to install rust and cargo on  manylinux2010 and musllinux_1_1, an alternative may be to simply require pydantic v1 on these old platforms.
